app/services/cclf/simple_bill_adapter.rb

```diff
@@ -0,0 +1,40 @@
+module CCLF
+  class SimpleBillAdapter
+    attr_reader :object
+
+    def initialize(object)
+      @object = object
+    end
+
+    def bill_type
+      raise 'Implement in sub-class'
+    end
+
+    def bill_subtype
+      raise 'Implement in sub-class'
+    end
+
+    def bill_scenario
+      case_type_adapter.bill_scenario
+    end
+
+    private
+
+    def case_type_adapter
+      @adapter ||= ::CCLF::CaseTypeAdapter.new(object.claim.case_type)
+    end
+
+    # delegate missing methods to object if it can respond
+    def method_missing(method, *args, &block)
```

Comment:
I was going to suggest if we could use [SimpleDelegator](http://ruby-doc.org/stdlib-2.4.2/libdoc/delegate/rdoc/SimpleDelegator.html) here to the same effect, but I've seen [some comments](https://github.com/rails/rails/issues/23824#issuecomment-221171899) by DHH on the subject.